x-pack/libbeat/common/aws/credentials.go

```diff
@@ -67,6 +74,27 @@ func GetAWSCredentials(config ConfigAWS) (awssdk.Config, error) {
 		return awsConfig, nil
 	}
 
+	emptyWebIdentity := WebIdentityProvider{}
+	if config.WebIdentityProvider != emptyWebIdentity {
+		logger.Debug("Using web_identity_role for AWS credential")
+		awsConfig, err := external.LoadDefaultAWSConfig()
+		if err != nil {
+			return awsConfig, errors.Wrap(err, "external.LoadDefaultAWSConfig failed when using role_arn")
+		}
+
+		stsSvc := sts.New(awsConfig)
+		p := stscreds.NewWebIdentityRoleProvider(stsSvc, config.WebIdentityProvider.RoleArn, config.WebIdentityProvider.RoleSession, stscreds.IdentityTokenFile(config.WebIdentityProvider.TokenFilePath))
```

Comment:
@kaiyan-sheng is the `RoleSession` param mandatory ? Just being curious as I activated the feature on my Filebeat pod, and only `RoleArn` + `TokenFilePath` seemed to be passed to the pod by EKS :
```
[...]
    - name: AWS_ROLE_ARN
      value: arn:aws:iam::<HIDDEN>:role/<HIDDEN>
    - name: AWS_WEB_IDENTITY_TOKEN_FILE
      value: /var/run/secrets/eks.amazonaws.com/serviceaccount/token
[...]
```